app/models/auth_source.rb

```diff
@@ -40,6 +40,14 @@ def auth_method_name
     "Abstract"
   end
 
+  def ldap?
```

Comment:
I don't like this method. I think `can_update_username?` is good enough and each `AuthSource` should implement its own `can_update_username?` and here `can_update_username?` will return `true`.